app/models/host_status/status.rb

```diff
@@ -82,5 +83,9 @@ def update_timestamp
     def update_status
       self.status = to_status
     end
+
+    def ensure_type_present
+      throw(:abort) if type.nil?
```

Comment:
That feels a bit heavy handed, wouldn't a regular validation work?

Reply:
Probably? I am trying to fail it silently but at this point I am just playing the guessing game. Maybe we should consider enforcing not null in the database after all. It could potentially create a lot of work for us in the immediate future, but right now I am just trying to come with ways to go around the root issue, and that might bite us in the leg as well.